Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -443,12 +443,13 @@ bool FullscreenManager::willEnterFullscreen(Element& element)
     m_pendingFullscreenElement = nullptr;
     m_fullscreenElement = &element;
 
-    document().resolveStyle(Document::ResolveStyleType::Rebuild);
-
-    Element* ancestor = m_fullscreenElement.get();
+    Element* ancestor = &element;
     do {
         ancestor->setFullscreenFlag(true);
 
+        if (ancestor == &element)
+            document().resolveStyle(Document::ResolveStyleType::Rebuild);
```

Comment:
Wonder if Rebuild is really needed.

Reply:
We unfortunately still do, for the 2 reasons I've mentioned in the commit message. I can look into removing it later though.